src/actions/bridgeWriteContract.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  type PublicClient,
+  Chain,
+  Transport,
+  encodeFunctionData,
+  WriteContractParameters,
+  Account,
+  Abi,
+  EncodeFunctionDataParameters,
+} from 'viem'
+import { writeContract } from 'viem/actions'
+import { l1CrossDomainMessengerABI } from '@eth-optimism/contracts-ts'
+import { OpChainL2, OpChainL1 } from '@roninjin10/rollup-chains'
+
+export async function bridgeWriteContract<
+  TAbi extends Abi | readonly unknown[] = Abi,
+  TFunctionName extends string = string,
+  TChainL1 extends OpChainL1 | undefined = OpChainL1,
+  TChainL2 extends OpChainL2 | undefined = OpChainL2,
+  TAccount extends Account | undefined = Account | undefined,
+  TChainOverride extends Chain | undefined = Chain | undefined,
```

Comment:
confused why we need this an L1 Chain

Reply:
We need every generic viem is using so we can reuse WriteContractParameters. this allows us to not have to write types that need to be kept in sync with viem